packages/frontend/webpack.config.js

```diff
@@ -65,6 +66,23 @@ module.exports = (env) => ({
     ],
   },
   plugins: [
+    {
+      apply: (compiler) =>
+        compiler.hooks.watchRun.tapAsync('watchRun', async (params, cb) => {
+          const currentBuild = (
+            await fs.readFile(path.join(__dirname, 'src/buildnum.js'))
+          ).toString()
+          const [, num] = currentBuild.match(/export default (\d+)/)
+          await fs.writeFile(
+            path.join(__dirname, 'src/buildnum.js'),
+            `export default ${+num + 1}`
```

Comment:
why is the `buildnum` 17 and why do we add 1 to it?

Reply:
The build number is the number of times the application bundle has been recompiled. When I was changing the config and testing the ui it changed `17` times while i was testing the feature.

From a devops perspective it's a way of visually labeling individual builds (like commit hashes). Using a simple integer is easier, and more meaningful, than a hash. It's an indicator of change _and_ rate of change.

I made a branch `explorer0` that deploys to [`explorer0.unirep.io`](https://explorer0.unirep.io/). You can use this branch like a dev branch and deploy changes there for review before merging to `main`.